bridges/TorrentGalaxyBridge.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+class TorrentGalaxyBridge extends BridgeAbstract {
+
+	const NAME = 'Torrent Galaxy Bridge';
+	const URI = 'https://torrentgalaxy.to';
+	const DESCRIPTION = 'Returns latest torrents';
+	const MAINTAINER = 'GregThib';
+	const CACHE_TIMEOUT = 14400; // 24h = 86400s
+
+	const PARAMETERS = array(
+		array(
+			'search' => array(
+				'name' => 'search',
+				'required' => true,
+				'title' => 'Type your query'
+			),
+			'lang' => array(
+				'name' => 'language',
+				'type' => 'list',
+				'exampleValue' => 'All languages',
+				'title' => 'Select your language',
+				'values' => array(
+					'All languages' => '0',
+					'English' => '1',
+					'French' => '2',
+					'German' => '3',
+					'Italian' => '4',
+					'Japanese' => '5',
+					'Spanish' => '6',
+					'Russian' => '7',
+					'Hindi' => '8',
+					'Other / Multiple' => '9',
+					'Korean' => '10',
+					'Danish' => '11',
+					'Norwegian' => '12',
+					'Dutch' => '13',
+					'Manderin' => '14',
+					'Portuguese' => '15',
+					'Bengali' => '16',
+					'Polish' => '17',
+					'Turkish' => '18',
+					'Telugu' => '19',
+					'Urdu' => '20',
+					'Arabic' => '21',
+					'Swedish' => '22',
+					'Romanian' => '23'
+				)
+			)
+		)
+	);
+
+	public function collectData(){
+		$url = self::URI . '/torrents.php' .
+			'?search=' . urlencode($this->getInput('search')) .
+			'&lang=' . $this->getInput('lang') .
+			'$sort=id&order=desc';
+		$html = getSimpleHTMLDOM($url)
+			or returnServerError('Error requiring the server.' . curl_error());
+
+		foreach($html->find('div.tgxtablerow') as $result) {
+			$identity = $result->find('div.tgxtablecell', 3)->find('div a', 0);
+			$authorid = $result->find('div.tgxtablecell', 6)->find('a', 0);
+			$creadate = $result->find('div.tgxtablecell', 11)->plaintext;
+			$glxlinks = $result->find('div.tgxtablecell', 4);
+
+			$item = array();
+			$item['uri'] = self::URI . $identity->href;
+			$item['title'] = $identity->plaintext;
+			$item['timestamp'] = DateTime::createFromFormat('d/m/y H:i', $creadate)->format('U');
+			$item['author'] = $authorid->plaintext;
+			$item['content'] = 
+				'<!DOCTYPE html><html><body>' .
+				'<h1>' . $identity->plaintext . '</h1>' .
+				'<h2>Links</h2>' .
+				'<p><a href="' . $glxlinks->find('a', 1)->href . '" title="magnet link">magnet</a></p>' .
+				'<p><a href="' . $glxlinks->find('a', 0)->href . '" title="torrent link">torrent</a></p>' .
+				'<h2>Infos</h2>' .
+				'<p>Size: ' . $result->find('div.tgxtablecell', 7)->plaintext . '</p>' .
+				'<p>Added by: <a href="' . $authorid->href . '" title="author profile">' . $authorid->plaintext . '</a></p>' .
+				'<p>Upload time: ' . $creadate . '</p>' .
+				'</body></html>';
```

Comment:
For this I suggest using the [heredoc](https://www.php.net/manual/en/language.types.string.php#language.types.string.syntax.heredoc) syntax. Also, do **not** include the doctype, as the content will be placed inside the content area of a feed (essentially inside a `div`). RSS-Bridge will produce the HTML document for you if you choose the HTML format.

```php
			$item['content'] = <<<HTML
<h1>{$identity->plaintext}</h1>
<h2>Links</h2>
<p><a href="{$glxlinks->find('a', 1)->href}" title="magnet link">magnet</a></p>
<p><a href="{$glxlinks->find('a', 0)->href}" title="torrent link">torrent</a></p>
<h2>Infos</h2>
<p>Size: {$result->find('div.tgxtablecell', 7)->plaintext}</p>
<p>Added by: <a href="{$authorid->href}" title="author profile">{$authorid->plaintext}</a></p>
<p>Upload time: {$creadate}</p>
HTML; // this must not be indented!
```

Disclaimer: I haven't tested it!

Reply:
Haaaaaaa! Great! Thanks alot for the trick. I'll change that.